### DISCUSSION TITLE
Links to helpful code and other resources for Lab 1

### DISCUSSION THREAD

**Discussion by eeholmes** [atsa-es/fish550-2023#6] [General]
In this thread I'll add links to code that you might find helpful for Lab 1

This repo folder has the Rmd files from the prior year Box-Jenkins lab. This was homework (problem set) based but might have code and examples that will help you with your analyses, forecasts and diagnostics.

https://github.com/atsa-es/atsa/tree/master/Labs/Week%202

The code in chapter 4 and 5 in the lab book will also be helpful

https://atsa-es.github.io/atsa-labs/chap-tslab.html
https://atsa-es.github.io/atsa-labs/chap-boxjenkins-.html

**Comment by eeholmes** [comment]
Your team members
https://github.com/orgs/atsa-es/teams/atsa-spring-2023

**Comment by eeholmes** [comment]
How to test for accuracy with training and test data
https://otexts.com/fpp2/accuracy.html

```
library(tidyverse)
library(forecast)
ruggerone_data <- readRDS(here::here("Lab-1", "Data_Images", "ruggerone_data.rds"))
dat <- ruggerone_data %>%  filter(species=="pink" & region=="ci") %>% mutate(log.returns = log(returns)) %>% select(year, log.returns)
dat <- ts(dat$log.returns, start=dat$year[1])
train <- window(dat, start=1961, end=1980)
test <- window(dat, start=1981, end=1985)
fit <- forecast::auto.arima(train)
accuracy(fit) # fit within the train data
accuracy(forecast(fit, h=5), test) # comparison of a forecast (5 year) to the test data (5 year)
```

**Reply by emmats** [reply]
I'm having trouble running the last line of code here. Error in forecast(fit.final, h = 5) : could not find function "forecast"

**Comment by eeholmes** [comment]
Ah, add library(forecast)

**Reply by emmats** [reply]
Ha! I did try forecast::accuracy(forecast(... But I will explicitly load the library.

**Comment by eeholmes** [comment]
@emmats Dealing with cycles in the sockeye (and other spp)

As you discovered, salmon are cyclic. This is due to their life-history. The models you are fitting do not take cycles into account. In this case, `diff(sockeye.ts)` reveals some strong correlation at 7 years. A bit odd as sockeye would tend to have 4 or 5 year cycles but this is summing over many regions.  For the lab, I would say just ignore the problems that you found with autocorrelated residuals. Discuss that you found the autocorrelation. But then go ahead with your forecasting tests.

If you wanted to address the cycling, you could include it explicitly, Here is code that shows you how:
https://github.com/atsa-es/fish550-2023/blob/main/Lab-1/Team-3/EEH-for-Emma.R

But for the lab, you don't need to worry about the cycling. Part of what you are learning in this lab, is the limitations of an ARIMA approach for real data. But you might also have better results for single regions where the lag structure (autocorrelation) in the data is stronger and the AR model can capture that. Here the 7-year cycle is weak and the strength is changing.

If you can't get enough of cyclic salmon (!), you can read ahead for material we will cover in a few weeks.
https://atsa-es.github.io/atsa-labs/chap-cyclic-sockeye.html

**Reply by emmats** [reply]
Thank you!
